aea/agent.py

```diff
@@ -226,22 +250,10 @@ def _run_main_loop(self) -> None:
         :return: None
         """
         logger.info("[{}]: Start processing messages...".format(self.name))
-        while not self.liveness.is_stopped:
-            self._tick += 1
-            self._spin_main_loop()
+        self._main_loop = SyncAgentLoop(self)
+        self._main_loop.start()
```

Comment:
extract as its own method similar to AEA for symmetry?

Reply:
```
def _run_main_loop(self) -> None:
        """
        Run the main loop of the agent.
        :return: None
        """
        logger.info("[{}]: Start processing messages...".format(self.name))
        self._main_loop.start()  # type: ignore
        logger.debug("[{}]: Exiting main loop...".format(self.name))
```

Reply:
method already extracted

Reply:
Yes in AEA, but not here. I am saying, let's do the same here..